crates/wasi-http/tests/all/p2.rs

```diff
@@ -67,7 +67,17 @@ impl WasiHttpView for Ctx {
         if let Some(send_request) = self.send_request.clone() {
             Ok(send_request(request, config))
         } else {
-            Ok(types::default_send_request(request, config))
+            #[cfg(feature = "default-send-request")]
+            {
+                Ok(types::default_send_request(request, config))
+            }
+            #[cfg(not(feature = "default-send-request"))]
+            {
+                Err(ErrorCode::InternalError(Some(
+                    "default-send-request feature disabled".to_string(),
+                ))
+                .into())
+            }
```

Comment:
FWIW we generally do not test all possible combinations of features, so I think it's reasonable to leave this out and have it be a compile error if the feature is disabled and tested. My thinking is along the lines of my above comment where I'd prefer to avoid reachable always-return-error functions if possible